lib/chef/provider/package/windows/exe.rb

```diff
@@ -55,17 +55,27 @@ def package_version
 
           def install_package
             logger.trace("#{new_resource} installing #{new_resource.installer_type} package '#{new_resource.source}'")
-            shell_out!(
-              [
-                "start",
-                "\"\"",
-                "/wait",
-                "\"#{new_resource.source}\"",
-                unattended_flags,
-                expand_options(new_resource.options),
-                "& exit %%%%ERRORLEVEL%%%%",
-              ].join(" "), timeout: new_resource.timeout, returns: new_resource.returns
-            )
+            begin
+              shell_out!(
+                [
+                  "start",
+                  "\"\"",
+                  "/wait",
+                  "\"#{new_resource.source}\"",
+                  unattended_flags,
+                  expand_options(new_resource.options),
+                  "& exit %%%%ERRORLEVEL%%%%",
+                ].join(" "), timeout: new_resource.timeout, returns: new_resource.returns
+              )
+            rescue Mixlib::ShellOut::ShellCommandFailed
+              if sensitive?
+                ex = Mixlib::ShellOut::ShellCommandFailed.new("Command execution failed. STDOUT/STDERR suppressed for sensitive resource")
+                raise ex
+              else
+                raise
+              end
```

Comment:
this is actually buggy and will still throw as part of the exception call chain that helpfully gets logged in newer rubies...

https://github.com/chef/chef/blob/master/lib/chef/provider/execute.rb#L59-L70

that is the correct code to use, and the fact that we're copypasta'ing this around suggests to me that this needs to get included in the Chef::Resource class directly so that any resource with sensitive set suppresses ShellCommandFailed exceptions correctly...

Reply:
Or it should get moved into mixlib-shellout as a feature there.

Reply:
Yep sensitive flag in mixlib-shellout itself would be good as well, then the initial exception with the sensitive info never gets thrown, so we don't have to worry about the exception chaining case.

Reply:
@lamont-granquist this got moved to mixlib-shellout. Can you re-review since you've got an ❌ on the PR?